### ISSUE TITLE
[HOLD for payment 2023-09-07] [$1000] Line does not break in 'Description' of Request money

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19872]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any chat
2. Go to request money
3. On description write something long
## Expected Result: 
The line should break to a new line at some point
## Actual Result: 
There is no line break

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.20-3 
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

[screen-capture (5).webm](https://github.com/Expensify/App/assets/43996225/9b070286-d556-4d28-863f-4bde8f177a4c)

https://github.com/Expensify/App/assets/43996225/24430830-dad9-4c78-b7f8-ee18c5170dad

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Habtamu-Asefa
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685338853319439

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010bfd86b76bb12ca6</li>
        <li>Upwork Job ID: 1665796276188585984</li>
        <li>Last Price Increase: 2023-06-19</li>
    </ul>
</details>

**Comment by therealsujitk** [comment]
@akinwale if I'm not mistaken you're working on a similar issue?

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Line does not break in 'Description' of Request money.
### What is the root cause of that problem?
The description input is not multiline input.
### What changes do you think we should make in order to solve the problem?
Make the description input multiline by add the following props
```javascript
      multiline
      textAlignVertical="top"
      numberOfLines={6}
```

### What alternative solutions did you explore? (Optional)

**Comment by akinwale** [comment]
> @akinwale if I'm not mistaken you're working on a similar issue?

@therealsujitk Yes, that is correct, but it's for a different area of the app, task descriptions.

**Comment by easeplan** [comment]
Has any solved this problem yet? I am asking because I want to try to fix it.

**Comment by Habtamu-Asefa** [comment]
Contributor details
Your Expensify account email: nazrihabtish@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0130c45e8d1f512bf8

**Comment by Oluwaniphise** [comment]
I guess it the line didn't break because the individual didn't add whitespace between the text. HTML just took it as one. Add a proper description with whitespace, it should work.

**Comment by Oluwaniphise** [comment]
Your Expensify account email: oduyaleenoch@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0170f6bd3ff8e0835d

**Comment by Habtamu-Asefa** [comment]
> I guess it the line didn't break because the individual didn't add whitespace between the text. HTML just took it as one. Add a proper description with whitespace, it should work.

I don't think that is the case. You can add as many whitespaces as you want but the line is not breaking.

**Comment by tjferriss** [comment]
>@akinwale if I'm not mistaken you're working on a similar issue?

@akinwale is it possible we can combine these issues? Can you share the link to the other issue?

**Comment by akinwale** [comment]
> > @akinwale if I'm not mistaken you're working on a similar issue?
> 
> @akinwale is it possible we can combine these issues? Can you share the link to the other issue?

@tjferriss I suppose this is possible, and I'm not opposed to this idea, but I'm not sure about all the details of how this combination would work re: the timeline, speed bonus, bounty (2x?) etc. I already have a PR open, and I can easily add the changes required for this in addition to making modifications to the MoneyRequestHeader component to support displaying multiline descriptions. You can follow the [PR discussion](https://github.com/Expensify/App/pull/19377) for more details as I imagine the same set of challenges may also apply here.

Additionally, there is a backend issue with saving task descriptions which contain newline characters. While I haven't tested yet, it may be possible that this could also happen with money requests.

**Comment by Habtamu-Asefa** [comment]
Does this mean it will not be considered as a bug for the tester?

**Comment by sobitneupane** [comment]
@akinwale @tjferriss 

Allowing new lines involve change in preview as well. We should also discuss about change in preview before moving forward. Following are the two places (preview and report) which will be impacted by the change. 
<img width="1042" alt="Screenshot 2023-06-05 at 14 08 30" src="https://github.com/Expensify/App/assets/25876548/8d8889ef-0b92-416e-8134-3fc2f83ae344">
<img width="1069" alt="Screenshot 2023-06-05 at 14 08 14" src="https://github.com/Expensify/App/assets/25876548/0b890b8a-cee0-4bc7-bd11-51cc3a6cd27e">

In [this issue](https://github.com/Expensify/App/issues/19135), we are going to allow adding new lines to description for task assignment. We don't show description in task preview. So, we didn't have any issue in task preview. But for task report we decided to do nothing for now as task view is being redesigned as discussed [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1685087242060519) by @thienlnam.

cc: @youssef-lr

**Comment by tjferriss** [comment]
@akinwale sorry, I should have been more specific in my question. My question was whether we can close this issue in favor of the other one. It seems like the answer is no, we should treat these separately.

**Comment by JavSFDev** [comment]
Contributor details
Your Expensify account email: javsfdev323@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01b7fc496875b2b3dd

**Comment by stitesExpensify** [comment]
@aimane-chnaif what do you think of this proposal? https://github.com/Expensify/App/issues/19872#issuecomment-1569427774

**Comment by tjferriss** [comment]
Discussing in Slack on how to move forward: https://expensify.slack.com/archives/C049HHMV9SM/p1686697408673289?thread_ts=1685338853.319439&cid=C049HHMV9SM

**Comment by tjferriss** [comment]
It looks like the consensus is to keep this issue separate from https://github.com/Expensify/App/issues/19389. Still awaiting proposals.

**Comment by ahmedGaber93** [comment]
Support multiline in `MenuItem`

prop types and default props.
```javascript
/** Boolean whether to display the title in multiline */
multiline: PropTypes.bool,

// defaultProps
multiline: false,
```

use styles.pre in `titleTextStyle` only if multiline false `!props.multiline && styles.pre`
https://github.com/Expensify/App/blob/9ed4a16e185ae43bdeb246a571173e6c0ec7a7df/src/components/MenuItem.js#L83

change title `numberOfLines` to be `numberOfLines={props.multiline ? undefined : 1}`
https://github.com/Expensify/App/blob/9ed4a16e185ae43bdeb246a571173e6c0ec7a7df/src/components/MenuItem.js#L193

![Screenshot 2023-06-15 at 10 52 17 AM](https://github.com/Expensify/App/assets/41129870/8e55ac04-8b31-4161-a7f6-31fafae95c5f)

**Comment by ahmedGaber93** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/19872#issuecomment-1569427774) to support multiline in `MenuItem`

**Comment by tjferriss** [comment]
@stitesExpensify @youssef-lr does @ahmedGaber93 proposal look good to you?

**Comment by ahmedGaber93** [comment]
>  let's move forward with the PR.

@youssef-lr @aimane-chnaif [PR](https://github.com/Expensify/App/pull/21664) is ready for review.

@youssef-lr I think you missed assigning me for this task.

**Comment by tjferriss** [comment]
It looks like we're still on hold.

**Comment by tjferriss** [comment]
Are we still waiting on the related PRs?

**Comment by tjferriss** [comment]
Still on hold

**Comment by tjferriss** [comment]
Easy Melvin, we're on hold.

**Comment by tjferriss** [comment]
Still on hold

**Comment by kassahun-asefa** [comment]
@tjferriss Line break seems to be working. Did we fix it as part of another PR?

**Comment by kassahun-asefa** [comment]
@tjferriss Line break seems to be working. Did we fix it as part of another PR?

**Comment by tjferriss** [comment]
@kassahun-asefa it doesn't appear to be working for me.

<img width="452" alt="Screenshot 2023-08-21 at 12 30 31" src="https://github.com/Expensify/App/assets/8292562/33980e30-174c-4fd4-8f92-742a5fa0c1fc">

**Comment by kassahun-asefa** [comment]
@tjferriss my bad I confused it with task description. It is not working yet.

**Comment by mountiny** [comment]
What are we holding on now? Could we add it to the title if possible?

**Comment by ahmedGaber93** [comment]
@tjferriss yes PR https://github.com/Expensify/App/pull/21664 is ready for review.

**Comment by tjferriss** [comment]
>This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

^From the PR, we're waiting on QA

**Comment by Habtamu-Asefa** [comment]
@tjferriss a kind reminder for payment

**Comment by Habtamu-Asefa** [comment]
@aimane-chnaif I don't understand. What do you mean it is a new feature? Does that mean there is no bonus for the reporter?

**Comment by Habtamu-Asefa** [comment]
@kavimuru @tjferriss It has been 4 days since the deadline for payment.

**Comment by tjferriss** [comment]
@Habtamu-Asefa sorry for the delay. Can you please apply to the job here? https://www.upwork.com/jobs/~011291928fb10896b6

**Comment by Habtamu-Asefa** [comment]
@tjferriss Thanks for the response. I have sent a proposal to the link you gave.

**Comment by Habtamu-Asefa** [comment]
@tjferriss checking on the status of my payment

**Comment by Habtamu-Asefa** [comment]
@tjferriss Nine days since the payment deadline. I look forward to no further delay.

**Comment by mallenexpensify** [comment]
Hired! @Habtamu-Asefa , can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~011291928fb10896b6

**Comment by mallenexpensify** [comment]
I applied the 50% bonus based on @aimane-chnaif 's timeline[ above](https://github.com/Expensify/App/issues/19872#issuecomment-1709718592)

Issue reporter: @Habtamu-Asefa paid $250 via Upwork
Contributor: @ahmedGaber93 paid $1500 via Upwork
Contributor+: @aimane-chnaif paid $1500 via Upwork

@aimane-chnaif can you please complete the BZ checklist above?

**Comment by mallenexpensify** [comment]
> BZ Checklist:
This is new feature rather than bug. Multiline description was not supported initially.

ah... I missed that.  We def need new regression tests for new features, otherwise we'll never be able to check/test them going forward.  Can you propose steps so I can create a TestRail GH @aimane-chnaif ?  Thx

**Comment by mallenexpensify** [comment]
Thanks @aimane-chnaif , closing! 
https://github.com/Expensify/Expensify/issues/318790
